ds3-autogen-c/src/main/java/com/spectralogic/ds3autogen/c/helpers/StructHelper.java

```diff
@@ -0,0 +1,253 @@
+        return indent(3) + "response->" + Helper.camelToUnderscore(structMember.getName()) + " = " + parserFunction + "\n";
+    }
+
+
+    public static String generateStructMemberArrayParserBlock(final String structResponseTypeName, final StructMember structMember) throws ParseException {
+        return indent(3) + "GPtrArray* " + structMember.getName() + "_array = _parse_" + structResponseTypeName + "(log, doc, child_node);\n"
+             + indent(3) + "response->" + structMember.getName() + " = (" + structMember.getType() + ")" + structMember.getName() + "_array->pdata;\n"
+             + indent(3) + "response->num_" + structMember.getName() + " = " + structMember.getName() + "_array->len;\n"
+             + indent(3) + "g_ptr_array_free(" + structMember.getName() + "_array, FALSE);\n";
+    }
+
+    public static String getParseStructMemberBlock(final String structName, final StructMember structMember) throws ParseException {
+        switch (structMember.getType()) {
+            case "ds3_str*":
+                return generateStructMemberParserLine(structMember, "xml_get_string(doc, child_node);");
+            case "double":
+            case "long":
+                return generateStructMemberParserLine(structMember, "xml_get_uint64(doc, child_node);");
+            case "int":
+                return generateStructMemberParserLine(structMember, "xml_get_uint16(doc, child_node);");
+            case "ds3_bool":
+                return generateStructMemberParserLine(structMember, "xml_get_bool(doc, child_node);");
+
+            default:
+                if (structMember.getType().endsWith("**")) { // array of another type
+                    return generateStructMemberArrayParserBlock(getResponseTypeName(structName), structMember);
+                }
+                return generateStructMemberParserLine(structMember, getParserFunctionName(structMember.getName()) + "(log, doc, child_node);");
+        }
+    }
+
+    public static String generateResponseParser(final String structName, final ImmutableList<StructMember> variables) throws ParseException {
+        final StringBuilder outputBuilder = new StringBuilder();
+
+        for (int structMemberIndex = 0; structMemberIndex < variables.size(); structMemberIndex++) {
+            outputBuilder.append(indent(2));
+
+            if (structMemberIndex > 0) {
+                outputBuilder.append("} else ");
+            }
+
+            final StructMember currentStructMember = variables.get(structMemberIndex);
+
+            outputBuilder.append("if (element_equal(child_node, \"").append(Helper.underscoreToCamel(currentStructMember.getName())).append("\")) {").append("\n");
+            outputBuilder.append(getParseStructMemberBlock(structName, currentStructMember));
+        }
+        // TODO Leaving the catch case commented out unless needed.
+        /*
+        outputBuilder.append(indent(3)).append("} else {").append("\n");
```

Comment:
We probably should leave this code in there, especially during testing to make sure the autogen code is generating the parsers correctly.